### DISCUSSION TITLE
Curious about your use of neovim tui

### DISCUSSION THREAD

**Discussion by aw1875** [amitds1997/remote-nvim.nvim#44] [Q&A]
I came across your project while working on a project that has some similarities. Wanted to say this project looks awesome and it's really impressive already! I wanted to shoot a question your way since it seems like you're doing something similar. I've been having issues getting Neovim's remote UI to work. For some reason, it just seems to flash and then nothing happens. I have attached a video to visualize what is going on. The left side is my local machine and the right side is an SSH connection into a container. Not sure why this is happening as it looks like you've taken the same approach:

Starting the ssh with pseudo terminal and port forwarding. In my example: `ssh host -t -L 5000:localhost:5000` (which is open on both).

Starting neovim `nvim --listen 0.0.0.0:5000` (not running headless yet but eventually plan to as well).
Connecting to remote `nvim --server localhost:5000 --remote-ui`.

If you faced any issues while working on this project like this I'd love to know how you solved it, I feel like I'm going crazy trying to figure this out.

https://github.com/amitds1997/remote-nvim.nvim/assets/4970015/b0507fa2-de1a-4820-9eca-1c7d907a69b7

---
~~Wanted to note the remote was running neovim version 0.9.2 and I'm running 0.9.1 locally if that could be it?~~

Edit: I'm stupid and 0.9.2 made some tui changes so it was incompatible. Sorry for the post.

**Comment by amitds1997** [comment]
I have been there more times than I would care to admit 😅. Let me know if there is something else I can help with 👍

**Reply by aw1875** [reply]
Appreciate it! I'm sure I'll reach out again at some point

**Reply by aw1875** [reply]
@amitds1997 Just wanted to give you a heads-up. Not sure if this is something you might encounter but if you're running a container on a newer Mac (aarch64) the appimage isn't recognized as a good executable format. It's kinda painful but building from source works fine. Again not sure if this is something you'll encounter but just in case.
